### PR TITLE
gui: Update homepage pool stats every 10 seconds

### PR DESCRIPTION
**Pull request by jholdstock**
Adds a websocket endpoint to the webserver which is used to push updates to connected clients every 10 seconds. Currently updating Last Payment Height, Last Work Height and Pool Hash Rate. More to come...

I suggest reviewing commit by commit.

Part of #96

### CODE REVIEW

File: gui/gui.go [hunk 1]
```diff
@@ -84,6 +84,30 @@ func (ui *GUI) route() {
...
+		if err != nil {
+			log.Error(err)
+			return
+		}
+
+		ui.SendUpdatedValues(poolStats)
+	}
+
+	// Use a ticker to push updates through the socket periodically
+	ticker := time.NewTicker(RefreshRate)
```
Comment (dnldd) [inline gui/gui.go]: let's have the goroutine exit cleanly via a context. Also lets limit the scope of the ticker to the goroutine it's used in:

```
go func(ctx context){
	ticker := time.NewTicker(RefreshRate)
	defer ticker.Stop()

	for {
		select{
		case <-ticker.C:
			socketUpdate()

		case ctx.Done():
			return
		}
	}
}()
```

File: gui/gui.go [hunk 2]
```diff
@@ -84,6 +84,30 @@ func (ui *GUI) route() {
 	ui.router.HandleFunc("/admin", ui.PostAdmin).Methods("POST")
 	ui.router.HandleFunc("/backup", ui.PostBackup).Methods("POST")
 	ui.router.HandleFunc("/logout", ui.PostLogout).Methods("POST")
+
+	// Websocket endpoint allows the GUI to receive updated values
+	ui.router.HandleFunc("/ws", ui.RegisterWebSocket).Methods("GET")
+
+	socketUpdate := func() {
```
Comment (dnldd) [inline gui/gui.go]: `socketUpdate` isn't being reused, doesn't need to be a anonymous function as a result. Let's fetch the stats and call `SendUpdatedValues` in the goroutine on a tick instead.

File: gui/websocket.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	},
+}
+
+type Message struct {
+	PoolHashRate      string `json:"poolhashrate"`
+	LastWorkHeight    uint32 `json:"lastworkheight"`
+	LastPaymentHeight uint32 `json:"lastpaymentheight"`
+}
+
+const RefreshRate = 5 * time.Second
```
Comment (dnldd) [inline gui/websocket.go]: `RefreshRate` should be unexported.

**Comment by jholdstock** [comment]
Implemented suggested changes and rebased onto master. Ready for a second review.

**Comment by dnldd** [comment]
Looks good.

**Pull request merged**